apps/web-connect/src/features/checkout/components/crossmint/CrossmintModal.tsx

```diff
@@ -64,30 +55,87 @@ const CrossmintModal: React.FC<CrossmintModalProps> = ({ isOpen, onClose, totalP
+                                        colors:{
+                                            text: styles.colors.textPrimary,
+                                        },
+                                    },
+                                    Input: {
+                                        font:{
+                                            size: styles.fontSizeBase,
+                                            weight: styles.fontWeightPrimary,
+                                        },
+                                        colors:{
+                                            text: styles.colors.textPrimary,
+                                            border: styles.colors.border,
+                                        },
+                                    },
+                                    PrimaryButton: {
+                                        font:{
+                                            size: styles.fontSizeBase,
+                                            weight: styles.fontWeightSecondary,
+                                        },
+                                        colors:{
+                                            background: styles.colors.backgroundTertiary,
+                                            text: styles.colors.textSecondary,
+                                        },
+                                    },
+                                },
+                                variables: {
+                                    colors: {
+                                        backgroundPrimary: styles.colors.background,
+                                        borderPrimary: styles.colors.border,
+                                        warning: styles.colors.danger,
+                                        textPrimary: styles.colors.textPrimary,
+                                        textSecondary: styles.colors.textSecondary,
+                                        danger: styles.colors.danger,
+                                        accent: styles.colors.accent,                                
+                                    },
+                                },
+                            }}
+                            recipient={{
+                                walletAddress: address as `0x${string}`
+                            }}
+                            lineItems={{
+                                collectionLocator: `crossmint:${collectionId}`,   
+                                callData: {
+                                    _amount: totalQty,
+                                    _to: address as `0x${string}`,
+                                    _promoCode: promoCode,
+                                    _expectedCostInUSDC: (BigInt(totalPriceInUsdc) / BigInt(10 ** 12)).toString(), // 10^12 to reduce 18 decimals to 6 decimals
+                                    totalPrice: formatWeiToEther(totalPriceInUsdc, 4), // convert to 4 decimal places for Crossmint
```

Comment:
This comment is misleading.
According to the [docs ](https://docs.crossmint.com/nft-checkout/advanced/usdc-support#validate-your-collection-meets-the-requirements) they will expect a USDC amount in "ETH" and (up to) 6 decimals.

Reply:
The price is only in ETH if the collection is denominated in ETH. I've adjusted the USDC price to 6 decimals

Reply:
ETH meaning the format here not the currency.

Reply:
I don't understand? The function `formatWeiToEther` takes an 18 decimal value and converts it to it's decimal value with a specific number of decimals. 

I need that functionality here.

Should I rename the utility function and update it every where it's called?
Or create a new utility function that does that same thing with a different name?

Not sure what the ideal fix you'd prefer is?

Reply:
No fix required here.
I wanted to clarify that ETH in this context is referring to the unit not the currency.
Crossmint expects usdc in ETH unit, since usdc has 6 decimals we can only expect up to 6 decimals after the `.`